main/define/src/mill/define/ModuleRef.scala

```diff
@@ -0,0 +1,9 @@
+package mill.define
+
+/**
+ * Used to refer to a module from another module without including the target
+ * module as a child-module of the first.
+ */
+case class ModuleRef[+T <: mill.define.Module](t: T) {
```

Comment:
Small suggestion, but could you be more meaningful with some of these names? For example in [mill-scip](https://github.com/ckipp01/mill-scip) I have some diffs now that look like

```diff
-              val zincWorker = sm.zincWorker.worker()
+              val zincWorker = sm.zincWorker.t.worker()
```
And it's really odd to have just `t` here.

Reply:
I think, the envisioned usage is to use the `def apply() = t`.

```diff
-              val zincWorker = sm.zincWorker.worker()
+              val zincWorker = sm.zincWorker().worker()
```

Reply:
Ahh duh,that makes more sense indeed.

Reply:
Shouldn't `t` be private then?

Reply:
Doesn't it invalidate the `case class` contract then? E.g. when pattern matching.

Reply:
this doesn't need to be a case class i think, just need a trait with an apply method